### PR TITLE
KAFKA-4500: code improvements

### PR DESCRIPTION
**Pull request by rekhajoshm**
- Removed redundant modifiers, not needed String.format()
- Removed unnecessary semicolon, additional assignment, inlined return 
- Added few param, return tags
- Using StringBuilder for consistency across codebase
- Using try-with-resources

### CODE REVIEW

**Comment by rekhajoshm** [comment]
@ijuma minor code changes, please review.thanks.

**Comment by rekhajoshm** [comment]
The jenkins test here passes for JDK 8 and Scala 2.12, but fails on JDK 8 and Scala 2.11.System issue?  @ijuma Please let me know if i need to do anything here? thanks.
screen shot 2016-12-10 at 11 22 03 am

**Comment by cmccabe** [comment]
Thanks for the pull request!

Can you separate out the JavaDoc fixes into a separate pull request?

I also am not sure that making some private things more visible is a good idea unless there is a reason to do so.

**Comment by rekhajoshm** [comment]
Thanks @cmccabe . Removed  javadoc changes. Note, I had only removed modifiers when they are implicit defaults, for eg: interface methods are default public, enum constructors are default private.
Please let me know what you think. thanks!

**Comment by rekhajoshm** [comment]
It passes on JDK 8, but states failed on streams test on JDK 7, though https://builds.apache.org/job/kafka-pr-jdk7-scala2.10/177/testReport/ shows all tests passed on JDK 7. Unrelated to this PR. Not sure if it was just an intermittent issue on jenkins? Can you please have a look @cmccabe ? Thanks.

File: clients/src/main/java/org/apache/kafka/common/security/authenticator/LoginManager.java [hunk 1]
```diff
@@ -61,7 +61,7 @@ private LoginManager(LoginType loginType, boolean hasKerberos, Map<String, ?> co
      *                  (i.e. consumer and producer)
      * @param configs configuration as key/value pairs
      */
-    public static final LoginManager acquireLoginManager(LoginType loginType, boolean hasKerberos, Map<String, ?> configs) throws IOException, LoginException {
+    public static LoginManager acquireLoginManager(LoginType loginType, boolean hasKerberos, Map<String, ?> configs) throws IOException, LoginException {
```
Comment (ijuma) [inline clients/src/main/java/org/apache/kafka/common/security/authenticator/LoginManager.java]: Why was this removed? Maybe an incorrect suggestion by an automated tool? See https://netbeans.org/bugzilla/show_bug.cgi?id=230827 for example.

File: tools/src/main/java/org/apache/kafka/tools/ThroughputThrottler.java [hunk 2]
```diff
@@ -42,9 +42,9 @@
     private static final long NS_PER_SEC = 1000 * NS_PER_MS;
     private static final long MIN_SLEEP_NS = 2 * NS_PER_MS;
 
-    long sleepTimeNs;
+    private long sleepTimeNs;
     long sleepDeficitNs = 0;
-    long targetThroughput = -1;
+    private long targetThroughput = -1;
     long startMs;
     private boolean wakeup = false;
```
Comment (ijuma) [inline tools/src/main/java/org/apache/kafka/tools/ThroughputThrottler.java:50]: I would restructure all these fields like:

```java
    private final long startMs;
    private final long sleepTimeNs;
    private final long targetThroughput;

    private long sleepDeficitNs = 0;
    private boolean wakeup = false;
```

File: tools/src/main/java/org/apache/kafka/tools/VerifiableLog4jAppender.java [hunk 3]
```diff
@@ -161,15 +160,10 @@ private static ArgumentParser argParser() {
      * we use VerifiableProducer from trunk tools package, and run it against 0.8.X.X kafka jars.
      * Since this method is not in Utils in the 0.8.X.X jars, we have to cheat a bit and duplicate.
      */
-    public static Properties loadProps(String filename) throws IOException, FileNotFoundException {
+    public static Properties loadProps(String filename) throws IOException {
```
Comment (ijuma) [inline tools/src/main/java/org/apache/kafka/tools/VerifiableLog4jAppender.java]: Why was `FileNotFoundException` removed from the signature? It can be useful to list specific subclasses that may be worth handling separately from the general `IOException`.

**Review by ijuma — CHANGES_REQUESTED** [review]
Thanks for the PR. Suggested a few minor improvements.

**Comment by rekhajoshm** [comment]
Thanks @ijuma for your review.updated.please have a look.thanks!

**Review by ijuma — APPROVED** [review]
LGTM, will merge after the tests pass.

**Comment by ijuma** [comment]
retest this please

**Comment by ijuma** [comment]
retest this please
